### PR TITLE
Add url shortener exercise

### PR DESCRIPTION
**Pull request by kosyfrances**
Exercise details gotten from https://github.com/gophercises/urlshort/blob/master/README.md

The major focus should be handler.go

### CODE REVIEW

File: urlshort/README.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# URL Shortener
+Exercise details from https://gophercises.com/exercises/urlshort
+
+The goal of this exercise is to create an [http.Handler](https://golang.org/pkg/net/http/#Handler) that will look at the path of any incoming web request and determine if it should redirect the user to a new page, much like URL shortener would.
```
Comment (dmigwi) [inline urlshort/README.md:4]: `create a` make more sense than `create an`.

File: urlshort/README.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
+# URL Shortener
+Exercise details from https://gophercises.com/exercises/urlshort
```
Comment (dmigwi) [inline urlshort/README.md:2]: How are you able to know that your code is correct without tests??

File: urlshort/README.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+func YAMLHandler(yaml []byte, fallback http.Handler) (http.HandlerFunc, error) {
+  parsedYaml, err := parseYAML(yaml)
+  if err != nil {
+    return nil, err
+  }
+  pathMap := buildMap(parsedYaml)
+  return MapHandler(pathMap, fallback), nil
+}
+```
+
```
Comment (dmigwi) [inline urlshort/README.md:26]: Your README is a bit confusing. Does it contain the description of the code you wrote or the actual task you were supposed to do? In the sample code above there is go routine created but your code does not have any go routines implementation.

File: urlshort/handler.go [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+	yaml "gopkg.in/yaml.v2"
+)
+
+// MapHandler will return an http.HandlerFunc (which also
+// implements http.Handler) that will attempt to map any
+// paths (keys in the map) to their corresponding URL (values
+// that each key in the map points to, in string format).
+// If the path is not provided in the map, then the fallback
+// http.Handler will be called instead.
+func MapHandler(pathsToUrls map[string]string, fallback http.Handler) http.HandlerFunc {
```
Comment (dmigwi) [inline urlshort/handler.go:15]: Great function commenting.

File: urlshort/main/main.go [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+	mapHandler := urlshort.MapHandler(pathsToUrls, mux)
+
+	//     // Build the YAMLHandler using the mapHandler as the
+	//     // fallback
+	yaml := `
+- path: /urlshort
+  url: https://github.com/gophercises/urlshort
+- path: /urlshort-final
+  url: https://github.com/gophercises/urlshort/tree/solution
+`
```
Comment (dmigwi) [inline urlshort/main/main.go:27]: Code could be better aligned.

File: urlshort/main/main.go [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+	yamlHandler, err := urlshort.YAMLHandler([]byte(yaml), mapHandler)
+	if err != nil {
+		panic(err)
+	}
```
Comment (dmigwi) [inline urlshort/main/main.go:31]: Unless you are running test only, kindly avoid `panic(err)`. It returns too much information which can affect the UX of your product.  Consider using 
```
if err != nil {
   fmt.Println(err)
   os.Exit(1)
}
``` 
exit with status 1 indicates an error occurred

**Review by dmigwi — COMMENTED** [review]
Great work though.

File: urlshort/README.md (hunk 2 above)
Comment (kosyfrances) [reply urlshort/README.md:2]: The exercise was just to implement handler.go as main.go was already provided

File: urlshort/README.md (hunk 3 above)
Comment (kosyfrances) [reply urlshort/README.md:26]: the README is just a description of the exercise

**Comment by kosyfrances** [comment]
Thanks @dmigwi

File: urlshort/main/main.go (hunk 6 above)
Comment (kosyfrances) [reply urlshort/main/main.go:31]: noted

**Pull request merged**